File: LC13.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> check;
        check['I']=1;
        check['V']=5;
        check['X']=10;
        check['L']=50;
        check['C']=100;
        check['D']=500;
        check['M']=1000;
        int pt=0,num=0;
        while(pt<s.size())
        {
            string tmp="";
            tmp+=s[pt];
            tmp+=s[pt+1];
            if(pt<s.size()-1 && (tmp=="CM" || tmp=="CD" || tmp=="XC" || tmp=="XL" || tmp=="IX" || tmp=="IV"))
            {
                num+=check[s[pt+1]]-check[s[pt]];
                pt++;
            }
            else
                num+=check[s[pt]];
            pt++;
        }
        return num;
    }
};

File: LC13.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        RN_dict = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        """
        #Principle of Roman numeral:
            I placed before V or X indicates one less, so four is IV (one less than five) and nine is IX (one less than ten)
            X placed before L or C indicates ten less, so forty is XL (ten less than fifty) and ninety is XC (ten less than a hundred) C placed before D or M indicates a hundred less, so four hundred is CD (a hundred less than five hundred and nine hundred is CM (a hundred less than a thousand)
        """
        sumRN = 0
        for i in range(0, len(s)-1):
            if RN_dict[s[i]] < RN_dict[s[i+1]]:
                sumRN -= RN_dict[s[i]]
            else:
                sumRN += RN_dict[s[i]]
        return sumRN+RN_dict[s[-1]]
        